src/main/java/net/sf/jabref/gui/groups/EntryTableTransferHandler.java

```diff
@@ -315,11 +315,9 @@ private boolean handleDraggedFilenames(String s, final int dropRow) {
      * @return success status for the operation
      */
     private boolean handleDraggedFiles(List<File> files, final int dropRow) {
-        final String[] fileNames = new String[files.size()];
-        int i = 0;
+        final List<String> fileNames = new ArrayList<>();
```

Comment:
I do not think they are harder to read, it is just some novel style, once you get used to it, it is plain simple and makes programming for fluent. In general I would prefer them if it makes sense.
The functional style makes it easier to crate a kind of  Collection-pipelline, e.g. introduce a filter, another transformation or limit the results or just replace the collector.  And could be easy parallelized, in cases where needed.